### PR TITLE
Jest test for all front end components except for login/register

### PR DESCRIPTION
**Pull request by B-Tanonaka**
1. Most components have 100% coverage. Outliers include fetch functions and setState functions in app.
2. Added additional fetch function that connects the camera to the AI model.
Screenshot 2024-01-23 at 5 42 03 PM

### CODE REVIEW

**Comment by JMStudiosJoe** [comment]
amazing

File: client/src/utils/fetchUserData.js [hunk 1]
```diff
@@ -46,6 +47,22 @@ export async function fetchProfileData(user) {
...
     console.error(err);
+    fetchLogOut();
+  }
+}
+
+export async function fetchImageToAI(image) {
+  try {
+    const formData = new FormData();
+    formData.append('image', image);
+    const res = await fetch('https://64ee-2601-646-c600-3560-d419-a27c-984f-1c5.ngrok-free.app/upload', {
```
Comment (JMStudiosJoe) [inline client/src/utils/fetchUserData.js:58]: is this an api key that should be an env var ?

**Review by JMStudiosJoe — COMMENTED** [review]
just one quick question before merging

File: client/src/utils/fetchUserData.js (hunk 1 above)
Comment (B-Tanonaka) [reply client/src/utils/fetchUserData.js]: That is a good call. I have moved it to an env file.

This is a temporary key that is going to a local machine and we will change the link once it is deployed.

**Review by JMStudiosJoe — APPROVED** [review]

**Pull request merged**